tests/unittests/graphOptzTest.cpp

```diff
@@ -2152,3 +2152,40 @@ TEST_F(GraphOptz, dceBeforeOptimizeTranpose) {
   // transpose should be eliminated and replaced by the transposed constant.
   EXPECT_TRUE(llvm::isa<Constant>(save1->getInput()));
 }
+
+TEST_F(GraphOptz, sinkTransposeBelowChannelShuffleNodes) {
+  const size_t inputDims[] = {3, 28, 28, 136};
+  Node *K;
+
+  K = mod_.createPlaceholder(ElemKind::FloatTy, inputDims, "input", false);
+  K = F_->createTranspose("unnecessary_transpose_1", K, {0, 3, 1, 2});
+  K = F_->createChannelShuffle("channel_shuffle", K, 4, 1);
+  K = F_->createTranspose("unnecessary_transpose_2", K, {0, 2, 3, 1});
+  auto *save = F_->createSave("ret", K);
+
+  EXPECT_EQ(F_->getNodes().size(), 6);
+
+  // Optimize away the unnecessary transposes
+  optimize(F_, CompilationMode::Infer);
+
+  // ensure the two unnecessary transposes a gone
```

Comment:
super nit: For comments everywhere, prefer capitalizing the first word and periods at the end of sentences :)